tests/unit/protocol-notification-bar.js

```diff
@@ -0,0 +1,57 @@
+/* global describe, beforeEach, it, spyOn */
+
+describe('protocol-notification.js', function() {
+
+    'use strict';
+
+    beforeEach(function () {
+        var details =
+            '<button class="mzp-c-button">Open Notification</button>';
+
+        document.documentElement.insertAdjacentHTML('afterbegin', details);
+    });
+
+    describe('interactions', function () {
+
+        var options = {
+            open: function () {}, // eslint-disable-line no-empty-function
+            close: function () {} // eslint-disable-line no-empty-function
+        };
+
+        beforeEach(function () {
+            spyOn(options, 'open');
+            spyOn(options, 'close');
+
+            var button = document.querySelector('.mzp-c-button');
+
+            button.addEventListener('click', function (e) {
+                e.preventDefault();
+
+                Mzp.Notification.init(button, {
+                    hasDismiss: true,
+                    onNotificationOpen: options.open,
+                    onNotificationClose: options.close
+                });
+            });
+
+            button.click();
+        });
+
+        it('opens as expected', function() {
+            expect(options.open).toHaveBeenCalled();
+
+            var notification = document.querySelector('.mzp-c-notification-bar');
+            expect(notification).toBeTruthy();
+        });
+
+        it('closes as expected', function() {
```

Comment:
Here you've created an ordered dependency on the "open" test needing to run before the "close" test, which will lead to intermittent failures. Tests run in a random order and not sequentially (hence why this passes sometimes and not at other times, like in CI). 

I'd suggest treating each test as it's own distinct piece of logic. Before each test runs you should create the notification, and after each test finishes you should remove the notification. That way, tests should pass regardless of the order they are run.